test/e2e/specs/editor/various/switch-to-draft.spec.js

```diff
@@ -0,0 +1,167 @@
+	await Promise.all( [
+		requestUtils.deleteAllPosts(),
+		requestUtils.deleteAllPages(),
+	] );
+} );
+
+test.afterAll( async ( { requestUtils } ) => {
+	await Promise.all( [
+		requestUtils.deleteAllPosts(),
+		requestUtils.deleteAllPages(),
+	] );
+} );
+
+test.describe(
+	'Clicking "Switch to draft" on a published/scheduled post/page',
+	() => {
+		[ 'schedule', 'publish' ].forEach( ( postStatus ) => {
+			[ 'large', 'small' ].forEach( ( viewport ) => {
+				[ 'post', 'page' ].forEach( ( postType ) => {
+					test( `should switch a ${ postStatus }-ed ${ postType } to draft in a ${ viewport } viewport`, async ( {
+						page,
+						switchToDraftUtils,
+						pageUtils,
+					} ) => {
+						await pageUtils.setBrowserViewport( viewport );
+
+						await switchToDraftUtils.createTestPost(
+							postType,
+							viewport,
+							postStatus === 'schedule'
+						);
+
+						await switchToDraftUtils.switchToDraftButton.click();
+
+						await page
+							.getByRole( 'dialog' )
+							.getByRole( 'button', { name: 'Cancel' } )
+							.click();
+
+						await expect
+							.poll(
+								switchToDraftUtils.getPostStatus,
+								`should leave a ${ postStatus }-ed ${ postType } ${ postStatus }-ed if canceled`
+							)
+							.toBe(
+								postStatus === 'schedule'
+									? 'future'
+									: postStatus
+							);
+
+						await switchToDraftUtils.switchToDraftButton.click();
+
+						await page
+							.getByRole( 'dialog' )
+							.getByRole( 'button', { name: 'OK' } )
+							.click();
+
+						await expect(
+							page.getByRole( 'button', {
+								name: 'Dismiss this notice',
+							} )
+						).toHaveText( `${ postType } reverted to draft.`, {
+							ignoreCase: true,
+						} );
+
+						await expect
+							.poll(
+								switchToDraftUtils.getPostStatus,
+								`should revert a ${ postStatus }-ed ${ postType } to a draft if confirmed`
+							)
+							.toBe( 'draft' );
+					} );
+				} );
+			} );
+		} );
+	}
+);
+
+class SwitchToDraftUtils {
+	/** @type {Page} */
+	#page;
+	/** @type {Admin} */
+	#admin;
+	/** @type {RequestUtils} */
+	#requestUtils;
+
+	constructor( { page, admin, requestUtils } ) {
+		this.#page = page;
+		this.#admin = admin;
+		this.#requestUtils = requestUtils;
+
+		this.switchToDraftButton = page
+			.getByRole( 'region', { name: 'Editor top bar' } )
+			.getByRole( 'button', { name: 'draft' } );
+	}
+
+	/**
+	 * Create a test post.
+	 *
+	 * @param {'post' | 'page'}   postType    The post type.
+	 * @param {'large' | 'small'} viewport    The viewport.
+	 * @param {boolean}           isScheduled Whether the post is scheduled.
+	 */
+	createTestPost = async ( postType, viewport, isScheduled = false ) => {
+		const payload = {
+			title: `Switch scheduled ${ postType } to draft`,
+			status: 'publish',
+			content: `<!-- wp:paragraph -->
+<p>This will be a scheduled ${ postType } edited in a ${ viewport } viewport</p>
+<!-- /wp:paragraph -->`,
```

Comment:
What do you think about making those more readable like this?

```suggestion
			content: [
				`<!-- wp:paragraph -->`,
				`<p>This will be a scheduled ${ postType } edited in a ${ viewport } viewport</p>`,
				`<!-- /wp:paragraph -->`,
			].join( '\n' ),
```

Reply:
I actually think this is less readable 😅 . It's only three lines so I think it's fine. For longer content, I agree your solution works better.  However, I think a better option might be to allow passing `blocks` to the `createPost`/`createPage` util and do the serializing automatically there. WDYT?

Reply:
Yep, that would work even better! 😄